src/main/java/org/minions/devfund/katerinaanzoleaga/WhackAMole.java

```diff
@@ -0,0 +1,211 @@
+                if (other.moleGrid[i][j] != this.moleGrid[i][j]) {
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        return Arrays.hashCode(this.moleGrid);
+    }
+
+    /**
+     * Places a mole in the row and column indicated.
+     * Returns false if it is not possible either because there is a mole there
+     * or when the row and column are invalid for the dimensions of the grid
+     * Also increases the number of moles left in the grid.
+     * @param x row
+     * @param y column
+     * @return boolean true if it was possible to place a mole there
+     */
+    boolean place(int x, int y) {
+        if (x < 0 || y < 0) {
+            return false;
+        }
+        int maxIndex = this.moleGrid.length - 1;
+        if (x <= maxIndex && y <= maxIndex && this.moleGrid[x][y] != 'M') {
+            this.moleGrid[x][y] = 'M';
+            this.molesLeft = this.molesLeft + 1;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Whacks a position indicated by the row and column parameter.
+     * If the position is out of the grid it will count as a failure.
+     * If there is a mole in the position, then the score will increase
+     * @param x row
+     * @param y column
+     */
+    void whack(int x, int y) {
+        if (x < 0 || y < 0) {
+            this.attemptsLeft = this.attemptsLeft - 1;
+        } else {
+            int maxIndex = this.moleGrid.length - 1;
+            if (x <= maxIndex && y <= maxIndex && this.moleGrid[x][y] == 'M') {
+                this.moleGrid[x][y] = 'W';
+                this.molesLeft = this.molesLeft - 1;
+                this.score = this.score + 1;
+            }
+            this.attemptsLeft = this.attemptsLeft - 1;
+        }
+    }
+
+    /**
+     * Prints the grid in user mode.
+     * This means without showing where the moles are,
+     * except the ones already whacked
+     */
+    void printGridToUser() {
+        int maxIndex = this.moleGrid.length - 1;
+        for (int i = 0; i <= maxIndex; i++) {
+            for (int j = 0; j <= maxIndex; j++)  {
+                if (this.moleGrid[i][j] == 'M' || this.moleGrid[i][j] == '\u0000') {
+                    System.out.print("   *");
+                } else {
+                    System.out.print("   " + this.moleGrid[i][j]);
+                }
+            }
+            System.out.println();
+        }
+    }
+
+    /**
+     * Prints the grid with the moles positions.
+     */
+    void printGrid() {
+        int maxIndex = this.moleGrid.length - 1;
+        for (int i = 0; i <= maxIndex; i++) {
+            for (int j = 0; j <= maxIndex; j++)  {
+                if (this.moleGrid[i][j] == '*') {
+                    System.out.print("   *");
+                } else {
+                    System.out.print("   " + this.moleGrid[i][j]);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)